governance/mixed.go

```diff
@@ -119,6 +145,62 @@ func (e *MixedEngine) assembleParams(headerParams *params.GovParamSet) *params.G
 	return p
 }
 
+func (e *MixedEngine) handleParamUpdate(old, new *params.GovParamSet) {
+	// NOTE: key set must be the same, which is guaranteed at NewMixedEngine
+	for k, oldval := range old.IntMap() {
+		if newval := new.MustGet(k); oldval != newval {
+			switch k {
+			// config.Istanbul
+			case params.Epoch:
+				e.config.Istanbul.Epoch = new.Epoch()
+			case params.Policy:
+				e.config.Istanbul.ProposerPolicy = new.Policy()
```

Comment:
not sure whether the locks can be removed.

```
func (bc *BlockChain) ProposerPolicy() uint64 {
	bc.chainConfigMu.RLock()
	defer bc.chainConfigMu.RUnlock()

	return bc.chainConfig.Istanbul.ProposerPolicy
}
```

Reply:
@jiseongnoh The rationale behind this is twofold.

(1)
Writing ChainConfig only takes place in `handleParamUpdate()`. All other accesses to ChainConfig are read accesses.
In the meantime, the call stack of `handleParamUpdate()` is rooted from `CreateSnapshot()`. I assumed that while `CreateSnapshot()` is running, there are no other goroutines that are simultaneously running along.

(2)
Previously, the lock is held in Blockchain object, not in ChainConfig object.
Since there are codes that are already reading ChainConfig directly without a read lock (such as below), I thought there's no point to introduce a lock here unless we're updating all ChainConfig accesses.
https://github.com/klaytn/klaytn/blob/5f18f2bf80f6fa039b232a359c7d801ef1a5e9e5/node/cn/handler.go#L314

Reply:
I hope there will be no problem with the locks.